src/jit/lclvars.cpp

```diff
@@ -3809,6 +3809,11 @@ void Compiler::lvaMarkLclRefs(GenTreePtr tree)
     allowStructs = varTypeIsStruct(varDsc);
 #endif // FEATURE_UNIX_AMD64_STRUCT_PASSING
 
+#if defined(FEATURE_SIMD) && defined(_TARGET_ARM64_)
+    // Allow mismatched TYP_STRUCT and TYP_SIMD* (since all TYP_SIMD* are Structs)
+    allowStructs = varTypeIsStruct(varDsc) && varTypeIsStruct(tree->gtType);
+#endif
```

Comment:
I'm not sure why you need this. This is a workaround for x64/ux to effectively bypass the type consistency validation below because the ABI handling introduces a case where it mixes `TYP_BYREF` and struct types. I believe that should be fixed, and we should definitely not do this on another target.

Reply:
There are type mismatches floating around.  Since AMD64 was ignoring, I wasn't sure whether ARM64 should ignore also.

I think I am fighting with #12590 which solved the AMD64 cases for `SIMD8` passed in integer register by inserting a `GT_BITCAST`. ARM64 case is not so easy.  With two argument registers used for passing one argument, GT_BITCAST feels wrong.  It would need to take a arg list as an operand I guess.

Right now, I am looking at resurrecting bits of the deleted code from #12590 to fix ARM64.

Reply:
> With two argument registers used for passing one argument, GT_BITCAST feels wrong.

Isn't this the case for GT_FIELD_LIST?

Reply:
While I see the arguments of a cal being morphed into a GT_FIELD_LIST, I do not see the call return being morphed to GT_FIELD_LIST.  I am looking at The `fgMorphCall()` monster now.

Reply:
Call returns don't use `GT_FIELD_LIST`. Currently when a call return is a struct returned in multiple registers, it is handled specially and must always assign to a lclVar.

Reply:
Took a look at "multi-reg-call-nodes.md".  `GT_STORE_MULTI_VAR` in document doesn't seem to exist.  My guess is that it is now `GT_STORE_LCL_VAR`

Reply:
> Took a look at "multi-reg-call-nodes.md". `GT_STORE_MULTI_VAR` in document doesn't seem to exist. My guess is that it is now `GT_STORE_LCL_VAR`

Wow, I'd forgotten that document existed. Yes, we have only implemented the Phase 1 implementation as described there. So, the temp to which the struct is copied must be on the stack.